rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/ecc/SW360EccService.java

```diff
@@ -0,0 +1,212 @@
+import org.jose4j.json.internal.json_simple.JSONArray;
+import org.jose4j.json.internal.json_simple.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Pageable;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.PagedModel;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Maps;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class SW360EccService {
+    @Value("${sw360.thrift-server-url:http://localhost:8080}")
+    private String thriftServerUrl;
+    
+    @NonNull
+    private final RestControllerHelper restControllerHelper;
+    
+    // ECC release view datatables, index of columns
+    private static final String RELEASE_DT_ROW_ECC_STATUS = "status";
+    private static final String RELEASE_DT_ROW_NAME = "name";
+    private static final String RELEASE_DT_ROW_VERSION = "version";
+    private static final String RELEASE_DT_ROW_GROUP = "group";
+    private static final String RELEASE_DT_ROW_ASSESSOR_CONTACT_PERSON = "assessor_contact_person";
+    private static final String RELEASE_DT_ROW_ASSESSOR_DEPARTMENT = "assessor_dept";
+    private static final String RELEASE_DT_ROW_ASSESSMENT_DATE = "assessment_date";
+
+    public JSONArray getECCPageData(User user,Pageable pageable) throws TException {
+        JSONArray jsonReleases = new JSONArray();
+        try {
+            PaginationData pageData = getPaginationData(pageable);
+            Map<PaginationData, List<Release>> releaseList = getReleaseList(user, pageData);
+            jsonReleases = getReleaseData(releaseList.values().iterator().next(), pageable);
+        } catch (Exception e) {
+            throw new TException(e.getMessage());
+        }
+        return jsonReleases;
+    }
+
+    private Map<PaginationData, List<Release>> getReleaseList(User user, PaginationData pageData) throws TException {
+        ComponentService.Iface client = getThriftComponentClient();
+        Map<PaginationData, List<Release>> releaseSummary = Maps.newHashMap();
+        try {
+            releaseSummary = client.getAccessibleReleasesWithPagination(user, pageData);
+        } catch (TException e) {
+            throw new TException(e.getMessage());
+        }
+        return releaseSummary;
+    }
+
+    private JSONArray getReleaseData(List<Release> releaseList, Pageable pageable) {
+        List<Release> sortedReleases = sortReleaseList(releaseList, pageable);
+        long count = getReleaseDataCount(pageable, releaseList.size());
+        final int start = 0;
+        JSONArray releaseData = new JSONArray();
+        for (int i = start; i < count; i++) {
+            JSONObject jsonObject = new JSONObject();
```

Comment:
No need to use jsonObject, instead use eccInformation field of releases after iterating the release list